### PR TITLE
Enable history server on batch nodes

### PR DESCRIPTION
**Pull request by paselem**
- [x] Pass through 18080 port in SSH for viewing history server
- [x] For local file system log paths, create the log path when the node starts up to prevent Spark from crashing when trying to run the history server

### CODE REVIEW

**Comment by jiata** [comment]
should we add the feature to url-ify the ssh command printlns  in this PR?
`
webui:              http://localhost:8080
history server: http://locahost:18080
...
`

**Comment by paselem** [comment]
@jiata I'd rather hold off - don't want to delay this. Can you open an issue for it?

File: node_scripts/install/spark.py [hunk 1]
```diff
@@ -159,6 +159,25 @@ def start_spark_master():
...
+
+
+def start_history_server():
+    # configure the history server
+    spark_event_log_enabled_key = 'spark.eventLog.enabled'
+    spark_event_log_directory_key = 'spark.eventLog.dir'
+    path_to_spark_defaults_conf = os.path.join(spark_home, 'conf/spark-defaults.conf')
+    properties = parse_configuration_file(path_to_spark_defaults_conf)
+
+    # only enable the history server if it was enabled in the configuration file
```
[the remark this answers is not in the document]
Comment (paselem) [reply node_scripts/install/spark.py]: Yes. I don't see another reason to turn it on if the user has not specified it and configured it. Any concerns with this?

[the remark this answers is not in the document]
Comment (jafreck) [reply node_scripts/install/spark.py]: No concerns, I actually think it's a better way than forcing the user to add a flag or something on the cli.

**Pull request merged**